test/test_read_all.py

```diff
@@ -127,22 +203,30 @@ def create_params():
 argnames, params, ids = create_params()
 
 
-def _get_read_fn(fmt, writing_library, reading_library):
-    fpath = DATA_DIR / f"{writing_library}{EXTENSIONS[fmt]}"
+def _get_read_fn(reading_library):
     read_fn = {
         "zarr": read_with_zarr,
         "pyn5": read_with_pyn5,
         "z5py": read_with_z5py,
         "zarrita": read_with_zarrita,
     }[reading_library]
-    return fpath, read_fn
+    return read_fn
 
 
 @pytest.mark.parametrize(argnames, params, ids=ids)
-def test_correct_read(fmt, writing_library, reading_library, codec):
+def test_correct_read(fmt, writing_library, reading_library, codec, nested,
+                      store_name, fpath):
+    if nested and reading_library == 'z5py':
+        pytest.skip("nested read not implemented in z5py")
+
     reference = imread(DATA_DIR / "reference_image.png")
-    fpath, read_fn = _get_read_fn(fmt, writing_library, reading_library)
-    test = read_fn(fpath, codec)
+    read_fn = _get_read_fn(reading_library)
+    if not os.path.exists(fpath):
+        raise RuntimeError(
+            f"file not found: {fpath}. Make sure you have generated the data "
+            "using 'make data'"
+        )
+    test = read_fn(fpath, codec, nested)
```

Comment:
While struggling with https://github.com/zarr-developers/zarr-python/issues/717, I introduced a [pseudo-checksum](https://github.com/ome/ome_zarr_test_suite/commit/1f13ac74feffc0a9392fd9155a1cd626f017d040#diff-c087348f9846c92c84a47fb9f99671d900427be8cbbee6583f00320f8d053d4cR15) in ome_zarr_test_suite that just asserts that there's at least one non-zero value in the opened dataset. My guess is that without https://github.com/zarr-developers/zarr-python/pull/716 that will be the case here as well.

Longer-term, I assume we could make use of some form of [true checksum'ing](https://github.com/zarr-developers/zarr-specs/issues/75) to show that when nesting is *not* properly detected, that the checksums differ (since all values revert to `fill_value`)

Reply:
Agreed that they are separate, but I think adding a checksum will cause test failures that we aren't seeing yet. But merging this for now. Thanks.